sklearn/metrics/_classification.py

```diff
@@ -1921,11 +1968,18 @@ class after being classified as negative. This is the case when the
             f"problems, got targets of type: {y_type}"
         )
 
+    if labels is None:
+        classes = np.unique(y_true)
```

Comment:
Here, we should probably call `_unique` instead just to be sure.
I assume that we could introduce a `pos_label` as well.